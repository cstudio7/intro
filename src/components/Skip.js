import React from 'react';
import {
    StyleSheet,
    Dimensions,
    TouchableHighlight,
    View,
    Text
} from 'react-native';
import PropTypes from 'prop-types';

const Skip = ({ ...props }) => {
    return (
        <View style={styles.container}>
            <TouchableHighlight
                onPress={() => props.onSkip()}
                underlayColor={props.lightBackground ? '#FAFAFA' : '#999'}
            >
                <Text style={[styles.text, {color: props.lightBackground ? '#000' : '#FAFAFA'}]}>
                    Skip
                </Text>
            </TouchableHighlight>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'flex-end',
        padding: 20
    },

    text: {
        fontSize: 16,
        fontWeight: 'bold'
    }
});

Skip.propTypes = {
    numberOfDots: PropTypes.number.isRequired,
    lightBackground: PropTypes.bool,
    onSkip: PropTypes.func
}


export default Skip;
