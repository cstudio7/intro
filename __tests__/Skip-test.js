import React from 'react';
import renderer from 'react-test-renderer';
import Skip from '../src/components/Skip.js';


test('Skip renders well', () => {
    const tree = renderer.create(<Skip />).toJSON();
    expect(tree).toMatchSnapshot()
})
