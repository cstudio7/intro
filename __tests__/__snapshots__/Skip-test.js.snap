// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skip renders well 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "padding": 20,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "color": "#FAFAFA",
          },
        ]
      }
    >
      Skip
    </Text>
  </View>
</View>
`;
